src/component/category-form/category-form.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import autoBind from '../../utils/utils';
+import './category-form.scss';
+
+const defaultState = {
+  title: '',
+  budget: '',
+};
+
+class CategoryForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = this.props.category || defaultState;
+    autoBind.call(this, CategoryForm);
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    this.props.onComplete(this.state);
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.setState({
+      [name]: value,
+    });
+  }
+  render() {
```

Comment:
Your app has that bug where if you make multiple categories (i.e. 3 categories) and you delete Category 2, the current props of that component do not update properly and the wrong information is displayed. Using the lifecycle hook ` static getDerivedStateFromProps` in this component should help with that. Reference this code here from lecture:
https://github.com/codefellows/seattle-javascript-401d23/blob/master/front-end/33-redux-middleware/kanban-combine-reducers/src/component/section-form/section-form.js